notification/service/notification.go

```diff
@@ -14,47 +14,79 @@ import (
 	"github.com/fabric8-services/fabric8-auth/rest"
+	"github.com/fabric8-services/fabric8-auth/test/configuration"
 	"github.com/fabric8-services/fabric8-auth/token/signer"
+	goaclient "github.com/goadesign/goa/client"
 	"github.com/goadesign/goa/uuid"
+	"net/http"
+	"sync"
 )
 
-// devNullNotificationService is the default dev service implementation. It does nothing.
-type devNullNotificationService struct{}
-
-func (s *devNullNotificationService) SendAsync(ctx context.Context, msg notification.Message) error {
-	return nil
-}
-
-func (s *devNullNotificationService) SendMessagesAsync(ctx context.Context, messages []notification.Message) error {
-	return nil
-}
-
 type notificationServiceImpl struct {
 	base.BaseService
 	config notification.Configuration
-	doer   rest.HttpDoer
 }
 
 // NewNotificationService creates a new service.
 func NewNotificationService(context servicecontext.ServiceContext, config notification.Configuration) service.NotificationService {
-	if config.GetNotificationServiceURL() == "" {
-		return &devNullNotificationService{}
-	}
 	return &notificationServiceImpl{
 		BaseService: base.NewBaseService(context),
 		config:      config,
-		doer:        rest.DefaultHttpDoer(),
 	}
 }
 
-// SendAsync creates a new goroutine and sends a message to fabric8-notification service
-func (s *notificationServiceImpl) SendAsync(ctx context.Context, msg notification.Message) error {
-	c, err := s.createClientWithContextSigner(ctx)
+// sendMessageAsync creates a new goroutine and sends a message to fabric8-notification service and wait till it responds, mainly used for testing
+func (s *notificationServiceImpl) sendMessageAsync(ctx context.Context, msg notification.Message, options ...configuration.HTTPClientOption) error {
+	c, err := s.createClientWithContextSigner(ctx, options...)
 	if err != nil {
 		return err
 	}
 
-	go s.send(ctx, c, msg)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		if e := s.send(ctx, c, msg); e != nil {
+			err = e
+		}
+	}()
+	wg.Wait()
+
+	return err
+}
+
+// sendMessagesAsync creates a new goroutine and sends multiple messages to fabric8-notification service and wait till it responds, mainly used for testing
+func (s *notificationServiceImpl) sendMessagesAsync(ctx context.Context, messages []notification.Message, options ...configuration.HTTPClientOption) []error {
+	c, err := s.createClientWithContextSigner(ctx, options...)
+	var errs []error
+	if err != nil {
+		errs = append(errs, err)
+		return errs
+	}
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for _, msg := range messages {
+			if e := s.send(ctx, c, msg); e != nil {
+				errs = append(errs, e)
+			}
+		}
+	}()
+	wg.Wait()
+
+	return errs
+}
+
+// SendMessageAsync creates a new goroutine and sends a message to fabric8-notification service
+func (s *notificationServiceImpl) SendMessageAsync(ctx context.Context, msg notification.Message) error {
```

Comment:
having twice the same method name with just a capital letter that differentiates is a bit confusing, IMO: `SendMessageAsync` vs `sendMessageAsync`. Why do we need 2 methods ?